src/content/insertable-streams/video-processing/js/webcodec-transform.js

```diff
@@ -0,0 +1,60 @@
+/*
+ *  Copyright (c) 2021 The WebRTC project authors. All Rights Reserved.
+ *
+ *  Use of this source code is governed by a BSD-style license
+ *  that can be found in the LICENSE file in the root of the source
+ *  tree.
+ */
+
+'use strict';
+
+/**
+ * Encodes and decodes frames using the WebCodec API.
+ * @implements {FrameTransform} in pipeline.js
+ */
+class WebCodecTransform { // eslint-disable-line no-unused-vars
+  constructor() {
+    // All fields are initialized in init()
+    this.debugPath_ = 'debug.pipeline.frameTransform_';
+  }
+  /** @override */
+  async init() {
+    console.log('[WebCodecTransform] Initializing encoder and decoder');
+    this.decoder_ = new VideoDecoder({ output: this.enqueueDecoded.bind(this), error: this.error });
+    this.encoder_ = new VideoEncoder({ output: this.decodeEncoded.bind(this), error: this.error });
+    this.encoder_.configure({codec: 'vp8', width: 640, height:480});
+    this.decoder_.configure({codec: 'vp8', width: 640, height:480});
+    this.controller_ = null;
```

Comment:
The Google style guide recommends all fields are declared in the constructor.

https://google.github.io/styleguide/jsguide.html#features-classes-fields

Reply:
Any particular preference for declaration ( = null) vs actual initialization in the constructor?
The other examples seemed to prefer to do init work that could fail in init().

Reply:
I'm not sure. Coming from C++ (minus exceptions), I have a habit of not doing much in the constructor.

The reason for having the init method was mainly to avoid instantiating WebGL on page load, since the WebGL transform is selected by default. In retrospect, I probably should have simplified things by avoiding an init method and using a callback/factory to instantiate things instead.